packages/chips/Chip.js

```diff
@@ -34,13 +36,21 @@ export class Chip extends Component {
         classList.delete(className);
```

Comment:
I don't think direct state mutation is the right solution. What about creating `new Set()`

Reply:
there could be other classes that we don't want to remove, which is what would happen if we created another Set.

Reply:
I think @mario-jerkovic means something like this:
```
const classList = new Set(this.state.classList);
```
@moog16 We should probably do this for all the other components too to avoid mutating state.

Reply:
Gotcha so we'd do:
```
const classList = new Set(this.state.classList);
classList.delete(className);
this.setState({classList}));
```

Reply:
This is very immutablejs-ish, are you saying this because React doesn't do a deep comparison of the Set's values?

Reply:
I was thinking about two potential problems with this approach. First is that this.setState is asynchronous and if you try to grab the value of this.state.classList it will be mutated before state update, and second is that it would be a pain to implement shouldComponentUpdate with mutable state.

Reply:
That makes sense. Potentially we could call removeClass/addClass more than once, and the classList might not be updated. I realize that it could update with incorrect classes. The code I wrote above also has this issue.

Reply:
Yes, that's right. And like you said, because React. js uses shallow comparison, if you just mute the state with a `delete` call on the state the component won't be re-rendered. I have made a simple example with `add/remove`
https://codesandbox.io/s/9y9pp9l7vy . 

At first glance I can tell you that you have to make it like this:
`
const classList = new Set([...this.state.classList]);
`

Reply:
Wow thanks for creating that codesandbox, really helpful. I like it, but question about this syntax:

```
this.setState(() => ({classList:newclassList}))
```

Why the function?

Reply:
No problem :) There is nothing wrong with just passing an object inside setState this is just another way of setting state and its more expressive. At the first glance you can see that the setState is asynchronous because of the callback and one added benefit is that function accepts the previous state and current props of the component so you can recalculate new state from the previous state.

Reply:
gotcha - @bonniezhou want to make this change? I will make an issue to update the other components.

Reply:
Done!